pkg/logstructured/sqllog/sql.go

```diff
@@ -283,10 +284,15 @@ func (s *SQLLog) postCompact() error {
 }
 
 func (s *SQLLog) CurrentRevision(ctx context.Context) (int64, error) {
-	if s.currentRev != 0 {
-		return s.currentRev, nil
+	currRev := s.currentRev.Load()
+	if currRev != 0 {
+		return currRev, nil
 	}
-	return s.d.CurrentRevision(ctx)
+	currRev, err := s.d.CurrentRevision(ctx)
+	if err == nil {
+		s.currentRev.Store(currRev)
```

Comment:
shouldn't this be a compare-and-swap, in case the other goroutine modified it since last loaded?